### ISSUE TITLE
Explicit Errors

### ISSUE THREAD

**Issue by disconsented** [SpongePowered/Sponge#257]
This idea is spawned from [this](https://forums.spongepowered.org/t/where-to-put-sponge-plugins/6933?u=disconsented).

In a nutshell when something goes wrong telling people what went wrong in plain English will help to improve usability for new/less technical users

**Comment by kenzierocks** [comment]
See #256 for a start on this sort of thing.

**Comment by Mumfrey** [comment]
I'm kind of making it an ongoing side project to add this kind of thing, which as @kenzierocks pointed out already spawned [this](https://github.com/SpongePowered/Sponge/commit/64292df89c93a3134eb36b8b20860bf876b19913) and #256. Obviously pushing the implementation forward is about balancing time spent on different areas of the codebase and adding this kind of friendly stuff can be usefully achieved in time slots that wouldn't be able to accommodate more involved coding tasks like lunch breaks etc.

I can't see this being solved in one fell swoop, there are many places that error handling would need to be sprinkled throughout the code base and I'm not sure that having a totally open-ended issue like this really helps, so it would seems sensible to deem it "accepted" but close it.

By the same token, if you can come up with a hit list of *x, y, z circumstances really need a friendly message as a priority* then update your issue description with a checklist and we can check them off as time allows. That would also give people finding this issue a more focused task to work on should they choose to do so.

If it's just going to be left as an open issue though, consider it implicitly "accepted" but we can close the actual issue.

**Comment by disconsented** [comment]
Excellent start, I hope to see more as the project progresses

**Comment by Mumfrey** [comment]
Closing this as it doesn't include specific points to be addressed and will effectively be "open" forever, since the intention is to make continual improvements to user-facing error feedback.
